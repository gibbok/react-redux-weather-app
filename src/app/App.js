import React from 'react'
import LocationFinderListContainer from '../locationFinder/locationFinderListContainer'

const App = () => (
  <div>
        App
        <LocationFinderListContainer />
  </div>
)

export default App
