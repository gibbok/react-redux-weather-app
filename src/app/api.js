import {apiUrl, apiKey, apiMapTemperature, apiMapPrecipitation, apiMapPressure, apiMapWind, apiMapCoulds} from './config'

export const find = query => `${apiUrl}find?q=${query}&APPID=${apiKey}`

export const weather = id => `${apiUrl}weather?id=${id}&APPID=${apiKey}&units=metric`

export const forecastDaily = id => `${apiUrl}forecast/daily?id=${id}&APPID=${apiKey}&units=metric`

export const forecast = id => `${apiUrl}forecast?id=${id}&APPID=${apiKey}&units=metric`

export const mapTemperature = () => `${apiMapTemperature}?APPID=${apiKey}`

export const mapPrecipitation = () => `${apiMapPrecipitation}?APPID=${apiKey}`

export const mapPressure = () => `${apiMapPressure}?APPID=${apiKey}`

export const mapWind = () => `${apiMapWind}?APPID=${apiKey}`

export const mapCloud = () => `${apiMapCoulds}?APPID=${apiKey}`
