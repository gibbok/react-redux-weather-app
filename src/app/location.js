// @flow
import {LocationType} from '../types'  // eslint-disable-line no-unused-vars

const location = ():LocationType => {
  return {
    locationId: 5128581, // nyc usa
    geo: [-74.006, 40.7143],
    isFetching: false
  }
}

export default location
