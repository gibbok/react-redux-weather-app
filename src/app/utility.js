// @flow
export const createMomentId = (timeInMs: number):string => {
  let moment = new Date(timeInMs)
  let day = moment.getUTCDay()
  let date = moment.getUTCDate()
  let month = moment.getUTCMonth()
  let year = moment.getUTCFullYear()
  return `${year}:${month}:${date}:${day}`
}
