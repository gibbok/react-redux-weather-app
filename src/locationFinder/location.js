const Location = ({onClick, name, country}) => (
  <li onClick={onClick}>
    {name} {country}
  </li>
)

export default Location
