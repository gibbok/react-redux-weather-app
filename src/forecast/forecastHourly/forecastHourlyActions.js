// @flow
import {ActionType} from '../../types' // eslint-disable-line no-unused-vars

import 'isomorphic-fetch'
import * as api from '../../app/api'
import * as types from './forecastHourlyActionTypes'

const getForecastHourly:ActionType = (query:number):ActionType => ({
  type: types.GET_FORECAST_HOURLY,
  payload: new Promise((resolve, reject) => {
    fetch(api.forecast(query)).then(response => {
      resolve(response.json())
    })
  })
})

const setForecastHourlyActiveReportType:ActionType = (type:string):ActionType => ({
  type: types.SET_FORECAST_HOURLY_ACTIVE_REPORT_TYPE,
  payload: type
})

export { getForecastHourly, setForecastHourlyActiveReportType }
