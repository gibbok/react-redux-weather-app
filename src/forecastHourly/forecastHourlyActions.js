import 'isomorphic-fetch'
import * as api from '../app/api'
import * as types from './forecastHourlyActionTypes'

const getForecastHourly = query => ({
  type: types.GET_FORECAST_HOURLY,
  payload: new Promise((resolve, reject) => {
    fetch(api.forecast(query)).then(response => {
      resolve(response.json())
    })
  })
})

const setForecastHourlyActiveReportType = type => ({
  type: types.SET_FORECAST_HOURLY_ACTIVE_REPORT_TYPE,
  payload: type
})

export { getForecastHourly, setForecastHourlyActiveReportType }
