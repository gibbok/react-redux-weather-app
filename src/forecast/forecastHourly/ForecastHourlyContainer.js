import { connect } from 'react-redux'
import ForecastHourly from './ForecastHourly'
import { getForecastHourly, setForecastHourlyActiveReportType } from './forecastHourlyActions'
import store from '../../app/store'

// const forecastHourly = (forecastHourly) => {
//   return forecastHourly.data.filter(x => x.momentId === forecastHourly.ui.selectedDate)
// }

const forecastHourly = (forecastHourly) => {
  return forecastHourly.data
}

const getForecastHourlyLocationId = () => {
  return store.getState().forecastDailyReducer.forecastDaily.app.locationId
}

const getActiveTypeReport = (state) => {
  return state.forecastHourlyReducer.forecastHourly.ui.activeReportType
}

const mapStateToProps = (state) => {
  return {
    forecastHourly: forecastHourly(state.forecastHourlyReducer.forecastHourly),
    activeTypeReport: getActiveTypeReport(state)
  }
}

const mapDispatchToProps = (dispatch, ownProps) => {
  return {
    onSummaryClick: (e) => {
      e.preventDefault()
      dispatch(setForecastHourlyActiveReportType('summary'))
    },
    onDetailsClick: (e) => {
      e.preventDefault()
      dispatch(setForecastHourlyActiveReportType('details'))
    }
  }
}

const ForecastHourlyContainer = connect(
  mapStateToProps,
  mapDispatchToProps
)(ForecastHourly)

store.dispatch(getForecastHourly(getForecastHourlyLocationId()))

export default ForecastHourlyContainer
