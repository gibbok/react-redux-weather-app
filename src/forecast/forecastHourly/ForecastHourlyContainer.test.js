// @flow
import React from 'react'
import { createMockStore, createMockDispatch } from 'redux-test-utils'
import {shallow} from 'enzyme'
import toJson from 'enzyme-to-json'
import configureStore from 'redux-mock-store'
import ForecastHourlyContainer from './ForecastHourlyContainer'
import {getForecastHourly, setForecastHourlyActiveReportType} from './forecastHourlyActions'
import location from '../../app/location'
import { createMomentId } from '../../app/utility'

describe('<ForecastHourlyContainer />', () => {
  it('should render', () => {
    const initialState = {
      forecastHourlyReducer: {
        forecastHourly: {
          data: [],
          app: {
            locationId: location().locationId, // nyc usa
            isFetching: false
          },
          ui: {
            selectedDate: createMomentId(Date.now()),
            activeReportType: 'summary'
          }
        }}
    }
    const mockStore = configureStore()
    const store = mockStore(initialState)
    const wrapper = shallow(<ForecastHourlyContainer store={store} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('should dispatch action getForecastHourly', () => {
    const state = 'state'
    const store = createMockStore(state)
    const action = getForecastHourly(location().locationId)

    store.dispatch(action)
    expect(store.getAction(action.type)).toEqual(action)
    expect(store.getActions()).toEqual([action])
    expect(store.isActionDispatched(action)).toBe(true)
    expect(store.isActionTypeDispatched(action.type)).toBe(true)
    expect(store.getState()).toBe(state)

    const dispatchMock = createMockDispatch()
    dispatchMock.dispatch(action)
    expect(dispatchMock.getAction(action.type)).toEqual(action)
    expect(dispatchMock.getActions()).toEqual([action])
    expect(dispatchMock.isActionDispatched(action)).toBe(true)
    expect(dispatchMock.isActionTypeDispatched(action.type)).toBe(true)
  })

  it('should dispatch action setForecastHourlyActiveReportType', () => {
    const state = 'state'
    const store = createMockStore(state)
    const action = setForecastHourlyActiveReportType('details')

    store.dispatch(action)
    expect(store.getAction(action.type)).toEqual(action)
    expect(store.getActions()).toEqual([action])
    expect(store.isActionDispatched(action)).toBe(true)
    expect(store.isActionTypeDispatched(action.type)).toBe(true)
    expect(store.getState()).toBe(state)

    const dispatchMock = createMockDispatch()
    dispatchMock.dispatch(action)
    expect(dispatchMock.getAction(action.type)).toEqual(action)
    expect(dispatchMock.getActions()).toEqual([action])
    expect(dispatchMock.isActionDispatched(action)).toBe(true)
    expect(dispatchMock.isActionTypeDispatched(action.type)).toBe(true)
  })
})
