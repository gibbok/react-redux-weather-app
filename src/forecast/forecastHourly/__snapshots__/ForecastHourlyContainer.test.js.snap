// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ForecastHourlyContainer /> should render 1`] = `
<ForecastHourly
  activeTypeReport="summary"
  forecastHourly={Array []}
  onDetailsClick={[Function]}
  onSummaryClick={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "notify": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": null,
    }
  }
/>
`;
